sql/template_xgboost.go

```diff
@@ -134,8 +134,11 @@ dpred = xgb_dataset('predict.txt', """{{.PredictionDatasetSQL}}""")
 bst = xgb.Booster({'nthread': 4})  # init model
 bst.load_model("{{.Save}}")  # load data
 preds = bst.predict(dpred)
-# TODO(typhoonzero): regression models may have different behavior
-pred_classes = np.argmax(np.array(preds), axis=1)
+
+# TODO(Yancey1989): using the train parameters to decide regressoin model or classifier model
```

Comment:
how to get the train parameters? Should we add `TrainIR` to `PredictIR`?

Reply:
Maybe we can save the TrainClause in the model struct: 
https://github.com/sql-machine-learning/sqlflow/blob/5ae82c045024e632a9d5eb898225c631187d688f/sql/model.go#L27-L30

And the prediction job can resolve the TrainClause to the parameters. And I think `PredictIR` is a good idea, do you want to resolve the Predict SQL and saved model into `PredictIR`?